### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by omnia487** [securefederatedai/openfederatedlearning#348] [Q&A] [answered]
Can OpenFl be used in cross-silos and cross-device scenarios?
Does OpenFl support both horizontal and vertical data partitioning?
What about the differential privacy support?
Does OpenFl support time-series datatype?

**Comment by alexey-gruzdev** [comment, marked as answer]
Hi @omnia487 ! Thanks for the interest in OpenFL.

OpenFL mostly covers cross-silos scenarios, but I wonder if you can describe specific cross-device use-cases. The terminology here is very vague, so it would be great to clarify what you exactly mean by cross-device. 
OpenFL currently supports only horizontal data partitioning. 
No DP example showcases  yet, however you can add noise to the data  inside `ShardDescriptor` on `Envoy` side. So, it should be pretty easy to integrate it.
For the time-series, I don't see any obstacles since our interfaces are pretty open to support this case as well, but out-of-the box we don't have such examples. Maybe, for right now,  [Next Word Prediction example](https://github.com/intel/openfl/tree/develop/openfl-tutorials/interactive_api/Tensorflow_Word_Prediction) could be a starting point for such cases.

**Reply by omnia487** [reply]
cross-device settings concern learning across user devices that have data created by a single user.

**Reply by alexey-gruzdev** [reply]
Okay, in that case, we don't support such cross-device scenarios yet.
